lang/src/accounts/program.rs

```diff
@@ -137,10 +125,22 @@ impl<'a, T: Id + Clone> Program<'a, T> {
     }
 }
 
-impl<'info, T> Accounts<'info> for Program<'info, T>
-where
-    T: Id + Clone,
-{
+impl<'a, T: Id> TryFrom<&AccountInfo<'a>> for Program<'a, T> {
```

Comment:
Just noticed that we're now implementing the `TryFrom` trait for `Program` and `Interface`. Any particular reason to start doing this? `Account` also has the `try_from_unchecked` method, so I guess it's not possible to also implement the trait there, at least not without splitting up into different implementation blocks.

Reply:
I used the `TryFrom` trait to make the function and its use more idiomatic, but I can certainly revert the change or add a `TryFrom` implementation to `Account` and `InterfaceAccount`, whatever you prefer! As you say, since there is `try_from_unchecked` in `Account`, I kept those in the same block for legibility.